cmd/skywire-cli/commands/config/gen.go

```diff
@@ -98,10 +201,158 @@ func init() {
+			if err == nil {
+				return b
+			}
+		}
+		return false
+	}
+	z, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; printf "%s"'`, skyenvfile, s)).String()
+	if err == nil {
+		b, err := strconv.ParseBool(z)
+		if err == nil {
+			return b
+		}
+	}
+
+	return false
+}
+
+func scriptExecArray(s string) string {
+	if visorconfig.OS == "windows" {
+		variable := s
+		if strings.Contains(variable, "[@]}") {
+			variable = strings.TrimRight(variable, "[@]}")
+			variable = strings.TrimRight(variable, "{")
+		}
+		out, err := script.Exec(fmt.Sprintf(`powershell -c '$SKYENV = "%s"; if ($SKYENV -ne "" -and (Test-Path $SKYENV)) { . $SKYENV }; foreach ($item in %s) { Write-Host $item }'`, skyenvfile, variable)).Slice()
+		if err == nil {
+			if len(out) != 0 {
+				return ""
+			}
+			return strings.Join(out, ",")
+		}
+	}
+	y, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; for _i in %s ; do echo "$_i" ; done'`, skyenvfile, s)).Slice()
+	if err == nil {
+		return strings.Join(y, ",")
+	}
+	return ""
+}
+
+func scriptExecInt(s string) int {
+	if visorconfig.OS == "windows" {
+		var variable string
+		if strings.Contains(s, ":-") {
+			parts := strings.SplitN(s, ":-", 2)
+			variable = parts[0] + "}"
+		} else {
+			variable = s
+		}
+		out, err := script.Exec(fmt.Sprintf(`powershell -c '$SKYENV = "%s"; if ($SKYENV -ne "" -and (Test-Path $SKYENV)) { . $SKYENV }; echo %s"`, skyenvfile, variable)).String()
+		if err == nil {
+			if (out == "") || (out == variable) {
+				return 0
+			}
+			i, err := strconv.Atoi(strings.TrimSpace(strings.TrimRight(out, "\n")))
+			if err == nil {
+				return i
+			}
+			return 0
+		}
+		return 0
+	}
+	z, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; printf "%s"'`, skyenvfile, s)).String()
+	if err == nil {
+		if z == "" {
+			return 0
+		}
+		i, err := strconv.Atoi(z)
+		if err == nil {
+			return i
+		}
+	}
+	return 0
+}
+
 var genConfigCmd = &cobra.Command{
 	Use:   "gen",
 	Short: "Generate a config file",
+	Long: func() string {
+		if visorconfig.OS == "linux" {
+			if skyenvfile == "" {
+				return `Generate a config file
+
+	Config defaults file may also be specified with
+	SKYENV=/path/to/skywire.conf skywire-cli config gen`
+			}
+			if _, err := os.Stat(skyenvfile); err == nil {
+				return `Generate a config file
+
+	skyenv file detected: ` + skyenvfile
+			}
+			return `Generate a config file
+
+	Config defaults file may also be specified with
+	SKYENV=/path/to/skywire.conf skywire-cli config gen`
+		}
+		return `Generate a config file`
+
+	}(),
 	PreRun: func(cmd *cobra.Command, _ []string) {
+		log := logger
+		if isEnvs {
+			if visorconfig.OS == "windows" {
+				envfile = envfileWindows
+			} else {
+				envfile = envfileLinux
+			}
+			fmt.Println(envfile)
```

Comment:
That looks like a debug print statement. Or what is the user supposed to be doing with the path that he set previously?